### PR TITLE
Adding numeric value for ordered list.

### PR DESCRIPTION
**Pull request by kalaspuffar**
Hi @bertfrees and @martinpub 

This PR is trying to solve issue #7

I've created a little XSLT script that adds numeric values to ordered lists, and it can handle nested lists. 

After a lot of trial and error, I've created a test script and inserted the function into the TTS generation in the pipeline.

The placement of the script might not be optimal, but it works.

Please review.

Best regards
Daniel

### CODE REVIEW

**Comment by bertfrees** [comment]
So [my help](https://github.com/daisy/pipeline/commit/168c6f16de4817c0b0f25e7c0211e28d11042a84) wasn't helpful at all?

This implementation seems to work for the most part, but I can see a few issues:
- The generated numbers are also present in the HTML, so this will probably result in double numbering in the visual rendering.
- The "plain" class that Martin mentioned in https://github.com/mtmse/pipeline/issues/7 will not be supported.

Of course, any numbering that deviates from the default (1., 2., 3.) will not be spoken correctly. I don't know if other kinds of numbering schemes exist in MTM publications though, so that might be OK.

Something else just caught my eye: are we sure that how nested lists are currently numbered ("2.1. Fåglarna hade tystnat och ...") is what Martin wants?

Finally a practical thing: I think it would be better if a separate test was created for this. It will make rebases easier.

**Comment by martinpub** [comment]
Thanks for this PR @kalaspuffar!

As @bertfrees has already pointed out, there are some issues still.

1. Yes, we need the "plain" class to be honoured in this regard.
2. Yes, as stated in the [Nordic Guidelines](https://format.mtm.se/nordic_epub/2020-1/#numbered-lists), even though not very common, we should expect the other numbering values available with `@type`.
3. `@start` should also be taken into account.
4. Regarding nested lists, I think "1." is appropriate for a sub-list item under the parent "2.", so Bert's observation that "2.1." is not ideal is correct.

Regarding double numbering, that we don't want, but I assumed this was something that was added in the TTS processing and then removed from the result, i.e. only there to feed the TTS engine with the text that it would otherwise miss out on?

**Comment by martinpub** [comment]
Me and @fredrikschill discussed this and agreed that my items 2-4 would be implemented.

1 should also be doable (it can be used as an indicator to prevent the job from even running on those lists), however we should note that it might be considered inappropriate as a general solution, as it is Nordic Guidelines-dependent. However, if all Nordic Guidelines-specific classes were put in the prefix "nordic:" ("nordic:plain"), it would reduce the risk of collision. This would involve a change in the guidelines however, so we should find a way to implement class "plain" while waiting for this.

Also, it is important to remove the generated numbers before outputing the resulting EPUB 3 TTS book.

**Comment by bertfrees** [comment]
A solution that does not take into account CSS may work for you but will never be general enough I'm afraid. 😞 

The "type" and "start" attributes are not standard HTML and I would also like to avoid taking into account "class" values (even if they're namespaced).

Even if I would take a less idealistic and more pragmatic approach and accept this Nordic specific handling in the main Pipeline code, then I would need to replace the implementation at some point anyway to correctly handle more general cases.

I'm very much convinced that the right way to do this is to make the script configurable through CSS rather than including custom processing in the code for everyone's needs. It's two birds with one stone because CSS also happens to be _the_ way to visually style lists in HTML/EPUB. And luckily, it's already largely implemented.

**Comment by martinpub** [comment]
I see @bertfrees. Given the timeframe and our prioritisations, we will go with the current solution for now (however improved according to the comments), fully aware that it will not be possible to integrate into the main version without refactoring it according to your proposed solution. Hopefully, we can return to this during the fall.

I was curious what you mean by "The 'type' and 'start' attributes are not standard HTML", by the way. Looking at the HTML5 spec, I don't find any indication that their status is weaker than other parts of the vocabulary?

**Comment by bertfrees** [comment]
Sorry, I was looking the the `li` element but they're attributes on the `ol` element :-)

**Comment by martinpub** [comment]
> Sorry, I was looking the the `li` element but they're attributes on the `ol` element :-)

OK, I see! :-)

**Comment by kalaspuffar** [comment]
Hi @martinpub and @bertfrees 

I've now done the changes in XSLT that Martin asked for, so now there is no structure in child lists, plain will not render any output, type and start is handled appropriately.

A bit annoying that the ```<xsl:number>``` tag format attribute didn't handle the use of a parameter but you can't get it all I guess.

Best regards
Daniel

File: modules/tts/epub3-tts/src/main/resources/xml/prepare-for-tts.xsl
Comment (bertfrees) [inline modules/tts/epub3-tts/src/main/resources/xml/prepare-for-tts.xsl]: Just use the `tts:speech-only` attribute (which is handled [here](https://github.com/daisy/pipeline/blob/ba10a7abf3612bc4df55b547ec3df7330def5b72/modules/tts/epub3-tts/src/main/resources/xml/tts-for-epub3.xpl#L283-L295)). Then you can be sure that nothing gets removed that shouldn't.

Comment (kalaspuffar) [reply modules/tts/epub3-tts/src/main/resources/xml/prepare-for-tts.xsl]: Hi @bertfrees 

Great suggestion, worked fine. Have changed to use that attribute instead.

Best regards
Daniel

**Comment by kalaspuffar** [comment]
Hi @bertfrees and @martinpub 

Still waiting for review of this PR.

Best regards
Daniel

**Comment by bertfrees** [comment]
It looks fine.

I would like to see the tests of all MTM specific features in a separate separate scenario or a xprocspec file. This is a general remark, not specific to this PR. It's to make it easier to do rebases or cherry-picks.

**Edit:** But, of course, I someone else does the rebases from now on, I don't care about the structure of the code.

**Review by martinpub — APPROVED** [review]

**Comment by bertfrees** [comment]
@martinpub For technical reasons (it's got to do with git-subrepo) it would have been better to merge https://github.com/mtmse/pipeline/pull/32 first before merging any other PRs. I've explained this to Daniel and Carl. No need to revert now, but something to keep in mind for the future.

**Comment by martinpub** [comment]
Thanks @bertfrees, I will revert this anyway, and pass the merge actions on to @kalaspuffar.

**Pull request merged**
